### PR TITLE
TEST PR

### PR DESCRIPTION
**Pull request by mvpopov**

### CODE REVIEW

**Comment by mvpopov** [comment]
TEST

**Comment by mvpopov** [comment]
Can one of the admins verify this patch?

**Comment by mvpopov** [comment]
TEST

**Comment by mvpopov** [comment]
TEST

**Comment by mvpopov** [comment]
TEST

**Comment by mvpopov** [comment]
ghjkl

**Comment by spbmvp** [comment]
TEST

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package ru.spbstu.telematics.popov.lab02;
+
+public class Introduction {
+    public static void main(String[] args) {
+        for(;;) {
+	    ;
+        }
+	while(true){
+		;
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Remove this empty statement. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

**Comment by mvpopov** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package ru.spbstu.telematics.popov.lab02;
+
+public class Introduction {
+    public static void main(String[] args) {
+	    //DoNothing;
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+	    System.out.println("Hello!");
+	for(;;) {
+	    ;
+        }
+	while(true){
+		;
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Remove this empty statement. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

**Comment by mvpopov** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by spbmvp** [comment]
Sonar

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	
+	void HhHhHhH() {
+	for (;;) {  // Noncompliant; end condition omitted
+		  // ...
+	}
+	int k=10;
+	for(int i=0; i<10; i++){
+		k++;
+	}
+	System.out.println(k);
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AS106)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	System.out.println("Hlksdf");
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AS106)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package ru.spbstu.telematics.popov.lab02;
+
+public class Introduction {
+    public static void main(String[] args) {
+	System.out.println("Hello!");
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AS106)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+	int k=10;
+	for(int i=0; i<10; i++){
+		k++;
+	}
+	System.out.println(k);
+	System.out.println("Hlksdf");
+	}
+	
+	
+	void test_divide() {
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AS00100)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
+package ru.spbstu.telematics.popov.lab02;
+
+public class Introduction {
+    public static void main(String[] args) {
+	System.out.println("Hello!");
+    }
+	
+	void HhHhHhH() {
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AS00100)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+	for (;;) {  // Noncompliant; end condition omitted
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: BLOCKER Add an end condition to this loop. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AS2189)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+	
+	void test_divide() {
+  int z = 0;
+  if (unknown()) {
+    // ..
+    z = 3;
+  } else {
+    // ..
+  }
+  z = 1 / z; // Noncompliant, possible division by zero
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: CRITICAL Make sure 'z' can't be zero before doing this division. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AS3518)
MAJOR Remove this useless assignment to local variable "z". [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AS1854)

**Comment by mvpopov** [comment]
SonarQube analysis reported 8 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 11]
```diff
@@ -0,0 +1,7 @@
+package ru.spbstu.telematics.popov.lab02;
+
+public class Introduction {
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AS1118)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 12]
```diff
@@ -0,0 +1,7 @@
...
+    public static void main(String[] args) {
+	System.err("Hello!");
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AS106)

**Comment by mvpopov** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by spbmvp** [comment]
Sonar

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 13]
```diff
@@ -0,0 +1,11 @@
...
+	    ////lknm,kjhbjjknml;,kjnbjnmk,
+	    while(true){
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: BLOCKER Add an end condition to this loop. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AS2189)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+		;
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Remove this empty statement. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

**Comment by mvpopov** [comment]
SonarQube analysis reported 4 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 11 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AS1118)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 12 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://172.17.0.4:9000/coding_rules#rule_key=squid%3AS106)

**Comment by mvpopov** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 15]
```diff
@@ -0,0 +1,10 @@
+package ru.spbstu.telematics.popov.lab02;
+
+public class Introduction {
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS1118)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 16]
```diff
@@ -0,0 +1,10 @@
...
+	//private Introduction() {
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 17]
```diff
@@ -0,0 +1,10 @@
...
+    	//	throw new IllegalAccessError("Utility class");
+  	//}
+    public static void main(String[] args) {
+	//logger.log("My Message");
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by mvpopov** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by spbmvp** [comment]
Sonar

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+	    
+	 ////lknm,kjhbjjknml;,kjnbjnmk,
+	while(true){
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: BLOCKER Add an end condition to this loop. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS2189)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+		;
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Remove this empty statement. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

**Comment by mvpopov** [comment]
SonarQube analysis reported 5 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by spbmvp** [comment]
Sonar

**Comment by spbmvp** [comment]
Sonar

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+
+public class Introduction {
+	private Introduction() {
+    		throw new IllegalAccessError("Utility class");
+  	}
+    public static void main(String[] args) {
+	logger.log("My Message");
+    }
+	
+	public void JhJHJh_21() {
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://172.20.0.199:9000/coding_rules#rule_key=squid%3AS00100)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 21]
```diff
@@ -0,0 +1,17 @@
...
+		//854wef4874swdf8
+		while(true) {
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: BLOCKER Add an end condition to this loop. [rule](http://172.20.0.199:9000/coding_rules#rule_key=squid%3AS2189)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 22]
```diff
@@ -0,0 +1,17 @@
...
+			;
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Remove this empty statement. [rule](http://172.20.0.199:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

**Comment by mvpopov** [comment]
SonarQube analysis reported 3 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 20 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS00100)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 21 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: BLOCKER Add an end condition to this loop. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS2189)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 22 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Remove this empty statement. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

**Comment by mvpopov** [comment]
SonarQube analysis reported 3 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 20 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS00100)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 23]
```diff
@@ -0,0 +1,17 @@
...
+	private Introduction() {
+    		throw new IllegalAccessError("Utility class");
+  	}
+    public static void main(String[] args) {
+	logger.log("My Message");
+    }
+	
+	public void JhJHJh_21() {
+		//854wef4874swdf
+		while(true) {
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: BLOCKER Add an end condition to this loop. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS2189)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 24]
```diff
@@ -0,0 +1,17 @@
...
+			;
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Remove this empty statement. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

**Comment by mvpopov** [comment]
SonarQube analysis reported 3 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 20 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS00100)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 23 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: BLOCKER Add an end condition to this loop. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS2189)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+    		throw new IllegalAccessError("Utility class");
+  	}
+    public static void main(String[] args) {
+	logger.log("My Message");
+    }
+	
+	public void JhJHJh_21() {
+		//854wef4874swdf
+		while(true) {
+			;//
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Remove this empty statement. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

**Comment by mvpopov** [comment]
SonarQube analysis reported 3 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 26]
```diff
@@ -0,0 +1,21 @@
+package ru.spbstu.telematics.popov.lab02;
+
+public class Introduction {
+	private Introduction() {
+    		throw new IllegalAccessError("Utility class");
+  	}
+    public static void main(String[] args) {
+	logger.log("My Message");
+	    int k= 0;
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Remove this unused "k" local variable. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS1481)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 27]
```diff
@@ -0,0 +1,21 @@
...
+	    while(true){
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: BLOCKER Add an end condition to this loop. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS2189)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 28]
```diff
@@ -0,0 +1,21 @@
...
+		    k++;
+	    }
+    }
+	
+	public void JhJHJh_21() {
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS00100)

**Comment by mvpopov** [comment]
SonarQube analysis reported 5 issues
* ![BLOCKER][BLOCKER] 2 blocker
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 29]
```diff
@@ -0,0 +1,21 @@
...
+	logger.log("My Message");
+	    int k= 0;
+	    while(true){
+		    k++;
+	    }
+    }
+	
+	public void hJHJh21() {
+		//854wef4874swdf
+		while(true) {
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: BLOCKER Add an end condition to this loop. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS2189)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 30]
```diff
@@ -0,0 +1,21 @@
...
+			;//
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Remove this empty statement. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

**Comment by mvpopov** [comment]
SonarQube analysis reported 4 issues
* ![BLOCKER][BLOCKER] 2 blocker
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 29 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: BLOCKER Add an end condition to this loop. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS2189)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java [hunk 31]
```diff
@@ -0,0 +1,21 @@
...
+	    int k= 0;
+	    while(true){
+		    k++;
+	    }
+    }
+	
+	public void hJHJh21() {
+		//854wef4874swdf
+		while(true) {
+			;
```
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Remove this empty statement. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 26 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Remove this unused "k" local variable. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS1481)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 27 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: BLOCKER Add an end condition to this loop. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS2189)

**Comment by mvpopov** [comment]
SonarQube analysis reported 4 issues
* ![BLOCKER][BLOCKER] 2 blocker
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 29 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: BLOCKER Add an end condition to this loop. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS2189)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 30 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Remove this empty statement. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 26 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: MINOR Remove this unused "k" local variable. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS1481)

File: Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java (hunk 27 above)
Comment (mvpopov) [inline Lab02/src/main/java/ru/spbstu/telematics/popov/lab02/Introduction.java]: BLOCKER Add an end condition to this loop. [rule](http://sonarqube:9000/coding_rules#rule_key=squid%3AS2189)

**Comment by mvpopov** [comment]
SonarQube analysis reported 4 issues
* ![BLOCKER][BLOCKER] 2 blocker
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Pull request closed without merging**